plugins/woocommerce/src/Internal/Admin/WcPaySubscriptionsPage.php

```diff
@@ -74,6 +71,67 @@ public function register_subscriptions_page() {
 		}
 	}
 
+	/**
+	 * Returns true if the store is eligible for the WooCommerce subsciptions empty state experiment.
+	 *
+	 * @return bool
+	 */
+	private function is_store_experiment_eligible() {
+		// Ineligible if WooCommerce Payments OR an existing subscriptions plugin is installed.
+		$installed_plugins      = PluginsHelper::get_installed_plugin_slugs();
+		$plugin_ineligible_list = array(
+			'woocommerce-payments',
+			'woocommerce-subscriptions',
+			'subscriptio',
+			'subscriptions-for-woocommerce',
+			'subscriptions-for-woocommerce-pro',
+			'sumosubscriptions',
+			'yith-woocommerce-subscription',
+			'xa-woocommerce-subscriptions',
+		);
+		foreach ( $plugin_ineligible_list as $plugin_slug ) {
+			if ( in_array( $plugin_slug, $installed_plugins, true ) ) {
+				return false;
+			}
+		}
+
+		// Ineligible if store address is not compatible with WCPay Subscriptions (US).
+		$store_base_location = wc_get_base_location();
+		if ( empty( $store_base_location['country'] ) || 'US' !== $store_base_location['country'] ) {
+			return false;
+		}
+
+		// Ineligible if store has not been active for at least 6 months.
+		if ( ! WCAdminHelper::is_wc_admin_active_in_date_range( 'month-6+' ) ) {
+			return false;
+		}
+
+		// Ineligible if store has not had any sales in the last 30 days.
+		if ( ! $this->get_store_recent_sales_eligibility() ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns true if the store has an order completed within the last 30 days.
+	 *
+	 * @return bool
+	 */
+	private function get_store_recent_sales_eligibility() {
+		// Get a single order that was marked as completed within the last 30 days.
+		$orders = wc_get_orders(
```

Comment:
FWIW this function call results in the following DB query:

```sql
SELECT wp_posts.ID
FROM wp_posts
INNER JOIN wp_postmeta
ON ( wp_posts.ID = wp_postmeta.post_id )
WHERE 1=1
AND ( ( wp_postmeta.meta_key = '_date_paid'
AND wp_postmeta.meta_value > '1649466738' ) )
AND wp_posts.post_type IN ('shop_order', 'shop_order_refund')
AND ((wp_posts.post_status = 'wc-pending'
OR wp_posts.post_status = 'wc-processing'
OR wp_posts.post_status = 'wc-on-hold'
OR wp_posts.post_status = 'wc-completed'
OR wp_posts.post_status = 'wc-cancelled'
OR wp_posts.post_status = 'wc-refunded'
OR wp_posts.post_status = 'wc-failed'))
GROUP BY wp_posts.ID
ORDER BY wp_posts.post_date DESC
LIMIT 0, 1
```

We should be mindful that this is essentially called on every admin page request (assuming the prior conditions are met).

In my local store with only a single order, this query took 0.5ms to execute.

IMO this is okay, as we're adding a "No thanks" button so the user will be able to dismiss this admin menu/page altogether and prevent this query from running.

/cc @james-allan

Reply:
Should we cache store eligibility in a transient so they only get this hit say ~once per day?

Reply:
Thanks @aprea and @haszari.

I've stored the result of this eligibility check as a transient: 3d68850.

I've updated the test step for **Store sales ineligibility** to detail how to delete this transient.